newIDE/app/src/Profile/CurrentUsageDisplayer.js

```diff
@@ -70,10 +127,13 @@ const CurrentUsageDisplayer = ({
         {usageRatio} in the past 24h.
       </Trans>
     );
+
+  const cannotUpgradeSubscription = !canUpgradeSubscription(subscription);
+
   return (
     <ColumnStackLayout noMargin>
       {hasSubscription ? (
-        !quota.limitReached ? (
+        remainingBuilds !== 0 ? (
```

Comment:
I think you should trust the back end and still use `quota.limitReached` on this